src/otx/core/data/entity/classification.py

```diff
@@ -74,3 +74,58 @@ def collate_fn(
 @dataclass
 class MulticlassClsBatchPredEntity(MulticlassClsBatchDataEntity, OTXBatchPredEntity):
     """Data entity to represent model output predictions for multi-class classification task."""
+
+
+@register_pytree_node
+@dataclass
+class MultilabelClsDataEntity(OTXDataEntity):
+    """Data entity for multi-label classification task.
+
+    :param labels: labels as integer indices
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.MULTI_LABEL_CLS
+
+    labels: LongTensor
```

Comment:
```suggestion
    :param labels: Multi labels represented as an one-hot vector.
    """

    @property
    def task(self) -> OTXTaskType:
        """OTX Task type definition."""
        return OTXTaskType.MULTI_LABEL_CLS

    labels: LongTensor
```
Can we make this as an one-hot vector?

Reply:
Could I ask you why we need to use the one-hot vector in this phase ? Since the `MultiLabelClsHead` conducts the label convert process, I think we don't need to do that from our side. https://github.com/open-mmlab/mmpretrain/blob/e95d9acb8919c59f7f1fab24a7701bdc23d25337/mmpretrain/models/heads/multi_label_cls_head.py#L99

Reply:
First of all, I saw that one-hot vector conversion is done in the evaluation step
https://github.com/openvinotoolkit/training_extensions/blob/6dfe06b9b28de932183bb70786a5efdf95f84ae0/src/otx/core/model/module/classification.py#L167
I think it would be better to be moved to the data pipeline for two perspectives: 1) reducing computation in the main thread and 2) clean code

About the link you gave, I can see [the very close line](https://github.com/open-mmlab/mmpretrain/blob/e95d9acb8919c59f7f1fab24a7701bdc23d25337/mmpretrain/models/heads/multi_label_cls_head.py#L96C16-L96C16) above you pointed seeks a one hot vector from `gt_score`. It seems no problem for me.

Reply:
Okay, then, it seems that we need to implement a function that calculates the number of classes to make a one-hot vector. A user already gives the `num_classes` to the model.head information, it's hard to use at the `Dataset` part. 

In my thoughts, this line https://github.com/openvinotoolkit/training_extensions/blob/6dfe06b9b28de932183bb70786a5efdf95f84ae0/src/otx/core/data/dataset/classification.py#L57 would be changed like below,

```python3
class OTXMultilabelClsDataset(OTXDataset[MultilabelClsDataEntity]):
    """OTXDataset class for multi-label classification task."""
    def __init__(***):
        super().__init__(***)
        self.num_classes = self._get_num_classes()

    def _get_num_classes(self):
        # find the max index of the labels by iterating the datumaro dataset
        # Or could we use the datumaro function to get the max label of the `Categories`?

    def _to_one_hot(self, label: Label):
        # Convert the label to the one-hot vector

    def _get_item_impl(self, index: int) -> MultilabelClsDataEntity | None:
        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
        img = item.media_as(Image)
        img_data, img_shape = self._get_img_data_and_shape(img)

        label_anns = [ann for ann in item.annotations if isinstance(ann, Label)]

        entity = MultilabelClsDataEntity(
            image=img_data,
            img_info=ImageInfo(
                img_idx=index,
                img_shape=img_shape,
                ori_shape=img_shape,
                pad_shape=img_shape,
                scale_factor=(1.0, 1.0),
            ),
            labels=torch.as_tensor([self._to_one_hot(ann.label) for ann in label_anns]),
        )

        return self._apply_transforms(entity)

    @property
    def collate_fn(self) -> Callable:
        """Collection function to collect MultilabelClsDataEntity into MultilabelClsBatchDataEntity in data loader."""
        return MultilabelClsBatchDataEntity.collate_fn
```

Or do we have any other option? Please correct me if I was wrong

Reply:
You may access `self.dm_subset.categories()` to get the label categories in Datumaro. It will allow you to get the number of total labels in the dataset I think.

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2739/commits/b6beac6aeb95277e4a8a95eb2026ea78f43f8687